core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -1413,6 +1436,41 @@ object ZManaged {
    */
   val never: ZManaged[Any, Nothing, Nothing] = ZManaged.fromEffect(ZIO.never)
 
+  /**
+   * Creates a scope in which resources can be safely preallocated.
+   */
+  final def scope[R]: ZManaged[R, Nothing, Preallocate] =
+    ZManaged {
+      Ref.make(Map.empty[Int, Exit[Any, Any] => ZIO[R, Nothing, Any]]).flatMap { finalizers =>
+        Ref.make(-1).map(_.update(_ + 1)).map { nextIndex =>
```

Comment:
```suggestion
        Ref.make(-1L).map(_.update(_ + 1L)).map { nextIndex =>
```

Reply:
I hope no one uses this in a way that this becomes necessary :D

Reply:
Haha. It's actually not that far fetched if you, for example, have a long running stream that opens and closes files in a high-throughput system.

Reply:
I was almost about to suggest `Map[UUID, Finalizer]` but I thought that really is overkill.

Reply:
we could use BigInteger :D